source/transactions-convenient-api/transactions-convenient-api.md

```diff
@@ -154,16 +167,32 @@ This method should perform the following sequence of actions:
 This method can be expressed by the following pseudo-code:
 
 ```typescript
+var BACKOFF_INITIAL = 1  // 1ms initial backoff
+var BACKOFF_MAX = 500  // 500ms max backoff
 withTransaction(callback, options) {
     // Note: drivers SHOULD use a monotonic clock to determine elapsed time
     var startTime = Date.now(); // milliseconds since Unix epoch
+    var retry = 0;
 
     retryTransaction: while (true) {
+        if (retry > 0) {
+            var backoff = Math.random() * min(BACKOFF_INITIAL * (1.25**retry), 
+                                              BACKOFF_MAX);
+            if (timeoutMS is None) {
+                timeoutMS = 120000
+            }
```

Comment:
suggest moving this up above the start of the withTransaction block; maybe something like `var TIMEOUT_MS = timeoutMS is None ? 120000 : TIMEOUT_MS`.

Then update the pseudocode to use TIMEOUT_MS instead of 120000 or timeoutMS.  

Also, not really in scope for your PR but this pseudocode doesn't account for timeoutMS (another CSOT miss?).  Can you update the blocks below as well to incorporate CSOT?  (there are a few places where 120000 is hard-coded).

Reply:
good catch on CSOT not properly being incorporated throughout, i just replaced all the 120 seconds to be TIMEOUT_MS in [bc9153e](https://github.com/mongodb/specifications/pull/1851/commits/bc9153eb4375e0b3a77ba8ecccd96af4432c3907)